pandas/core/apply.py

```diff
@@ -284,6 +284,14 @@ def transform_str_or_callable(self, func) -> DataFrame | Series:
         except Exception:
             return func(obj, *args, **kwargs)
 
+    def _filter_numeric_only(self) -> list[Any]:
+        if "numeric_only" in self.kwargs and self.kwargs["numeric_only"] is True:
```

Comment:
I don't believe there is any need for the `is True` here. This would give surprising results if the user passes e.g. `np.bool_(True)`.

Also, can you use `kwargs.get("numeric_only", False)`